### PR TITLE
Make cookie name configurable

### PR DESCRIPTION
**Pull request by aburgel**
/domain @rynonl @jmileham @samandmoore 
/cc @chrislopresto @superwomyn 

Goes with https://github.com/Betterment/test_track_js_client/pull/5

### CODE REVIEW

File: lib/test_track.rb [hunk 1]
```diff
@@ -57,8 +57,4 @@ def private_url
   def enabled?
     enabled_override.nil? ? !Rails.env.test? : enabled_override
   end
-
-  def fully_qualified_cookie_domain_enabled?
```
Comment (aburgel) [inline lib/test_track.rb]: This seemed very specific to the session cookie management, so I moved it into a private method in session.rb.

File: spec/dummy/config/environments/production.rb [hunk 2]
```diff
@@ -20,7 +20,7 @@
   # config.action_dispatch.rack_cache = true
 
   # Disable Rails's static asset server (Apache or nginx will already do this).
-  config.serve_static_assets = false
+  config.serve_static_files = false
```
Comment (aburgel) [inline spec/dummy/config/environments/production.rb]: Fixed some deprecation warnings when running the specs.

File: lib/test_track.rb (hunk 1 above)
Comment (superwomyn) [inline lib/test_track.rb]: Sounds good.

File: spec/models/test_track/session_spec.rb [hunk 3]
```diff
@@ -118,6 +118,18 @@
     end
 
     context "cookies" do
+      it "uses the default cookie name if not configured" do
```
Comment (superwomyn) [inline spec/models/test_track/session_spec.rb]: Should we have a test where the cookie name is configured? Or is that testing the framework?

File: spec/models/test_track/session_spec.rb [hunk 4]
```diff
@@ -118,6 +118,18 @@
...
+        subject.manage {}
+        expect(cookies).to include 'tt_visitor_id'
+      end
+
+      it "uses the configured cookie name when enabled" do
```
Comment (superwomyn) [inline spec/models/test_track/session_spec.rb]: nit: "... when configured" (rather than "enabled")

**Comment by superwomyn** [comment]
<< domain

**Comment by superwomyn** [comment]
couple of small comments & domainLGTM.

File: spec/test_track_spec.rb [hunk 5]
```diff
@@ -65,16 +65,4 @@
       end
     end
   end
-
-  describe ".fully_qualified_cookie_domain_enabled?" do
```
Comment (aburgel) [inline spec/test_track_spec.rb]: There's already coverage in session_spec.rb for when the flag is enabled, so i didn't think these tests added much.

Comment (superwomyn) [inline spec/test_track_spec.rb]: Doh! I didn't see them in the changes so I assumed they were missing. Sort of funny considering I'm the one who wrote them. #facepalm

File: spec/models/test_track/session_spec.rb [hunk 6]
```diff
@@ -118,12 +118,12 @@
...
 
     context "cookies" do
-      it "uses the default cookie name if not configured" do
+      it "uses the default cookie name when not configured" do
         subject.manage {}
         expect(cookies).to include 'tt_visitor_id'
       end
 
-      it "uses the configured cookie name when enabled" do
+      it "uses the custom cookie name when configured" do
```
Comment (superwomyn) [inline spec/models/test_track/session_spec.rb]: ❤️

**Comment by superwomyn** [comment]
:shipit:

**Pull request merged**